llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -766,6 +766,12 @@ NVPTXTargetLowering::NVPTXTargetLowering(const NVPTXTargetMachine &TM,
       AddPromotedToType(Op, MVT::bf16, MVT::f32);
   }
 
+  for (MVT VT : {MVT::i1, MVT::i16, MVT::i32, MVT::i64}) {
+    setOperationAction(
+        {ISD::SINT_TO_FP, ISD::UINT_TO_FP, ISD::FP_TO_SINT, ISD::FP_TO_UINT},
```

Comment:
Should we make it conditional on SM/PTX here, instead of checking in the custom lowering?

Reply:
Makes sense, done.

Reply:
It would be useful to update the review with the latest changes. I got puzzled for a bit to see this pull request closed with this item marked as done, but unchanged. And then I went to check the actual commit and find the expected changes to be present there.